### PR TITLE
Add 'ECONNREFUSED' to retrypolicyfilter

### PR DESCRIPTION
**Pull request by enlyt**
For some reason getting this reply after retrying several times. Requests seem to be working and going through after handling this.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@enlyt__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

**Comment by hasonmsft** [comment]
@enlyt , you PR looks good to me except the target branch. Could you re-send your PR to dev branch? Refer to the [CONTRIBUTING.md](https://github.com/Azure/azure-storage-node/blob/dev/CONTRIBUTING.md) for more detailed information. 

Thanks,
Haibo

File: lib/common/filters/retrypolicyfilter.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ RetryPolicyFilter._handle = function (self, requestOptions, next) {
         // we should NOT retry within the SDK as the stream data is not valid anymore if we retry directly.
         if (!returnObject.outputStreamSent && returnObject.error && azureutil.objectIsNull(returnObject.retryable) &&
              ((!azureutil.objectIsNull(returnObject.response) && retryInfo.retryable) || 
-               (returnObject.error.code === 'ETIMEDOUT' || returnObject.error.code === 'ESOCKETTIMEDOUT' || returnObject.error.code === 'ECONNRESET' || returnObject.error.code === 'EAI_AGAIN'))) {
+               (returnObject.error.code === 'ECONNREFUSED' || returnObject.error.code === 'ENOTFOUND' || returnObject.error.code === 'ETIMEDOUT' || returnObject.error.code === 'ESOCKETTIMEDOUT' || returnObject.error.code === 'ECONNRESET' || returnObject.error.code === 'EAI_AGAIN'))) {
```
Comment (hasonmsft) [inline lib/common/filters/retrypolicyfilter.js]: For ENOTFOUND, we should only retry if it happens for the secondary endpoint, for primary endpoint, we should NOT retry it.

**Comment by hasonmsft** [comment]
@enlyt , we manually take your PR and released in 1.4.0 release. Thanks for your contribution.

**Pull request closed without merging**